src/dm_ls_scf_methods.F

```diff
@@ -114,15 +119,23 @@ SUBROUTINE ls_scf_init_matrix_S(matrix_s, ls_scf_env)
 
       ! compute sqrt(S) and inv(sqrt(S))
       IF (ls_scf_env%use_s_sqrt) THEN
+
          CALL dbcsr_create(ls_scf_env%matrix_s_sqrt, template=ls_scf_env%matrix_s, &
                            matrix_type=dbcsr_type_no_symmetry)
          CALL dbcsr_create(ls_scf_env%matrix_s_sqrt_inv, template=ls_scf_env%matrix_s, &
                            matrix_type=dbcsr_type_no_symmetry)
-
-         CALL matrix_sqrt_Newton_Schulz(ls_scf_env%matrix_s_sqrt, ls_scf_env%matrix_s_sqrt_inv, &
-                                        ls_scf_env%matrix_s, ls_scf_env%eps_filter, &
-                                        ls_scf_env%sign_sqrt_order, &
-                                        ls_scf_env%eps_lanczos, ls_scf_env%max_iter_lanczos)
+         IF (ls_scf_env%s_sqrt_method .EQ. ls_s_sqrt_ns) THEN
```

Comment:
I think a `SELECT CASE` with a `DEFAULT CASE` is a better fit than `IF ELSE` here.